encryptio.py

```diff
@@ -1,55 +1,302 @@
+
+
+## mode execution
+def encrypt_mode(args):
+    match args.cryptoAlgorithm:
+        case CryptoAlgorithms.AES:
+            encrypt_aes(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.DES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES)
+        case CryptoAlgorithms.TRIPLEDES:
+            encrypt_des(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode, DES3)
+        case CryptoAlgorithms.IDEA:
+            encrypt_idea(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+        case CryptoAlgorithms.BLOWFISH:
+            encrypt_blowfish(str.encode(args.cryptoData), str.encode(args.cryptoKey), args.cryptoMode)
+
+
+# def decrypt_mode(args):
+
+# def bruteforce_mode(args):
+
+## mode listing
+def arg_encryption_mode(args):
+    encrypt_mode(args)
+
+
+def arg_decryption_mode(args):
+    encrypt_mode(args)
+
+
+# def arg_bruteforce_mode(args):
+#     bruteforce_mode(args)
+
+
 ## argument checker
 def check_key(key):
-
     if not (key.isalpha()):
         raise argparse.ArgumentTypeError('Key is not a string')
 
-    if(len(key) < 8): 
+    if len(key) < 8:
         raise argparse.ArgumentTypeError('Key to short - Minimum 8 characters')
 
     return key
-    
-## parse arguments 
-parser = argparse.ArgumentParser(description='encryptio.py is an easy to use generic encryption tool', epilog='--- encryptio.py - Moritz Nentwig Laura Tzigiannis', add_help=True)
 
-parser.add_argument('encryptionAlgorithm', help='Algorithm to encrypt', type=EncryptionAlgorithms, choices=EncryptionAlgorithms)
-parser.add_argument('encryptionKey', help='Key to encrypt with', type=check_key)
-parser.add_argument('encryptionLibrary', help='library to encrypt object with', type=str)
 
-parser.add_argument('-o', '--outputFile', help='file to write output to', type=argparse.FileType('w'), dest='outputFile')
+def create_binary_file(ciphertext, algorithm, additional_information):
+    # header: 8 bytes for algorithm, 1 byte for type of additional_information, 32 bytes for additional_information = 41 bytes
+    filename = 'encryptio_' + datetime.now().strftime("%H_%M_%S") + '.enc'
+
+    if additional_information:
+        information_padded = pad_data(additional_information[1], 32)
+    else:
+        information_padded = str.encode('0'*32)
+
+    with open(filename, 'wb') as f:
+        f.write(str.encode(algorithm))
+        if not additional_information:
+            f.write(str.encode('0'))
+        elif additional_information[0] == 'nonce_tag':
+            f.write(str.encode('T'))
+        elif additional_information[0] == 'nonce':
+            f.write(str.encode('N'))
+        elif additional_information[0] == 'initialization_vector':
+            f.write(str.encode('I'))
+        f.write(information_padded)
+        f.write(ciphertext)
+
+
+def pad_data(data_to_pad, block_size):
```

Comment:
Same as above. A Helpful Comment above the method name should provide information about what is done in the method. Have a look at the other methods.

Reply:
Done.